src/Persistence/Repository/EmployeeRepository.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Domain.Entity;
+using Domain.Repository;
+using Microsoft.EntityFrameworkCore;
+
+namespace Data.Repository;
+
+public class EmployeeRepository : IEmployeeRepository {
+    private readonly WriteDatabaseContext _context;
+
+    public EmployeeRepository(WriteDatabaseContext context) {
+        _context = context;
+    }
+
+    public async Task AddAsync(EmployeeEntity entity) {
+        await _context.Employees.AddAsync(entity);
+    }
+    
+    public async Task<EmployeeEntity?> GetByIdAsync(Guid id) {
+        return await _context.Employees.FindAsync(id);
+    }
+    
```

Comment:
Are you sure that findAsync includes all the salary history and workshifts for an employee ? Maybe it does and maybe it doesnt.  So, you probably need to Include and FirstOrDefaultAsync LINQ commands